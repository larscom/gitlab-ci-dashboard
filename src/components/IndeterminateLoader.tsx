import { MantineColor, MantineNumberSize, Progress } from '@mantine/core'

interface Props {
  color?: MantineColor
  size?: MantineNumberSize
}
export default function IndeterminateLoader({
  color = 'teal',
  size = 'md',
}: Props) {
  return <Progress color={color} animate size={size} value={100} />
}
