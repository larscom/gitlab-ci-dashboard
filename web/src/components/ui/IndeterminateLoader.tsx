import { MantineColor, MantineNumberSize, Progress } from '@mantine/core'

interface IndeterminateLoaderProps {
  color?: MantineColor
  size?: MantineNumberSize
}
export default function IndeterminateLoader({
  color = 'teal',
  size = 'md',
}: IndeterminateLoaderProps) {
  return <Progress color={color} animate size={size} value={100} />
}
