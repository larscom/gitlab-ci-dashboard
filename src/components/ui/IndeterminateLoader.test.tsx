import { render } from '@testing-library/react'

import IndeterminateLoader from './IndeterminateLoader'

describe('IndeterminateLoader', () => {
  it('should fully render', () => {
    const { container } = render(
      <IndeterminateLoader color="grape" size="xl" />
    )
    expect(container).toMatchSnapshot()
  })
})
