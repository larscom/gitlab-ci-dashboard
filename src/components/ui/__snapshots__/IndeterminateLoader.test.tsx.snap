// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`IndeterminateLoader > should fully render 1`] = `
<div>
  <div
    class="mantine-Progress-root mantine-cfbpgz"
  >
    <div
      aria-valuemax="100"
      aria-valuemin="0"
      aria-valuenow="100"
      class="mantine-18fgoy1 mantine-Progress-bar"
      role="progressbar"
      style="width: 100%;"
    />
  </div>
</div>
`;
